### PR TITLE
[TEST-328] Payara Samples fails to clean up after self

### PR DESCRIPTION
**Pull request by cubastanley**
Issue arises due to JAAS context created in initial run of test suite persisting across runs. This means that the first test case of RegisterLoginModuleTest will always fail after an initial run of the test suite. The change I've made does resolve this issue but does require a manual server restart after running as, for some reason, having this restart done in the code breaks the following tests.

### CODE REVIEW

File: samples/asadmin/src/test/java/fish/payara/samples/asadmin/RegisterLoginModuleTest.java [hunk 1]
```diff
@@ -62,6 +66,25 @@ public void successfulRegistrationUpdatesLoginConf() throws IOException {
         assertSuccess(result);
         String contents = loginConf();
         assertContains("test1 {", contents);
+        
+        result = asadmin("delete-auth-realm", "test1");
+        System.out.println(result.getOutput());
+        assertSuccess(result);
+        
+        if (contents.contains("test1 {")) {
```
Comment (pdudits) [inline samples/asadmin/src/test/java/fish/payara/samples/asadmin/RegisterLoginModuleTest.java]: If it didn't contain this line the assertion on line 68 would have failed. So this test is unnecessary.

File: samples/asadmin/src/test/java/fish/payara/samples/asadmin/RegisterLoginModuleTest.java [hunk 2]
```diff
@@ -62,6 +66,25 @@ public void successfulRegistrationUpdatesLoginConf() throws IOException {
...
+            String newFileContents = contents.replace(contents.substring(contents.indexOf("test1")), "");
```
Comment (pdudits) [inline samples/asadmin/src/test/java/fish/payara/samples/asadmin/RegisterLoginModuleTest.java]: You should remove everything up to `}` in login.conf.

Comment (cubastanley) [reply samples/asadmin/src/test/java/fish/payara/samples/asadmin/RegisterLoginModuleTest.java]: Pretty sure this does this. Takes a substring from test1 up to the end of the file as the entry test1 is the final entry in the file.

Comment (pdudits) [reply samples/asadmin/src/test/java/fish/payara/samples/asadmin/RegisterLoginModuleTest.java]: Right, I didn't realize it's until end of the string.

However, did you consider an option that we shouldn't check for success at line 68? Existing JAAS context will make the result a warning, but that's fine for us.

File: samples/asadmin/src/test/java/fish/payara/samples/asadmin/RegisterLoginModuleTest.java [hunk 3]
```diff
@@ -62,6 +66,23 @@ public void successfulRegistrationUpdatesLoginConf() throws IOException {
...
         assertContains("test1 {", contents);
+        
+        result = asadmin("delete-auth-realm", "test1");
+        System.out.println(result.getOutput());
+        assertSuccess(result);
+       
+        String newFileContents = contents.replace(contents.substring(contents.indexOf("test1")), "");
+        System.out.print(newFileContents);
+        try {
+            BufferedWriter writer = new BufferedWriter(new FileWriter(System.getProperty("java.security.auth.login.config")));
```
Comment (MarkWareham) [inline samples/asadmin/src/test/java/fish/payara/samples/asadmin/RegisterLoginModuleTest.java]: What is the purpose of this file?

Comment (cubastanley) [reply samples/asadmin/src/test/java/fish/payara/samples/asadmin/RegisterLoginModuleTest.java]: It's the login.conf file in the domain config folder in the server, need to remove the definition of JAAS context test1 from here in order for the test to pass on subsequent executions

File: samples/asadmin/src/test/java/fish/payara/samples/asadmin/RegisterLoginModuleTest.java [hunk 4]
```diff
@@ -62,6 +66,23 @@ public void successfulRegistrationUpdatesLoginConf() throws IOException {
...
+            writer.write(newFileContents);
+            writer.close();
```
Comment (AlanRoth) [inline samples/asadmin/src/test/java/fish/payara/samples/asadmin/RegisterLoginModuleTest.java]: Could go for a try with resources

File: samples/asadmin/src/test/java/fish/payara/samples/asadmin/RegisterLoginModuleTest.java [hunk 5]
```diff
@@ -62,6 +66,21 @@ public void successfulRegistrationUpdatesLoginConf() throws IOException {
         assertSuccess(result);
         String contents = loginConf();
         assertContains("test1 {", contents);
+        
+        result = asadmin("delete-auth-realm", "test1");
+        System.out.println(result.getOutput());
+        assertSuccess(result);
+       
+        String newFileContents = contents.replace(contents.substring(contents.indexOf("test1")), "");
```
Comment (pdudits) [inline samples/asadmin/src/test/java/fish/payara/samples/asadmin/RegisterLoginModuleTest.java]: All right, let's keep it this way then. But please make this modification more straightforward.
For example `contents.substring(0,context.indexOf(..))` is simpler. And also a comment, that you expect it to be last entry of login.conf.

**Pull request merged**